doc/migrations/6.3.1_6.x.y.md

```diff
@@ -36,3 +36,48 @@ It is also possible to completely disable client certificate verification during
 handshake in `just_tls` by setting `tls.verify_mode` to `none`.
 
 For more information regarding configuration of TLS for C2S see [Listener modules](../listeners/listen-c2s/#tls-options-for-c2s)
+
+## `exml` upgraded to 4.0
+
+### Change of internal format of XML messages affects the ways to upgrade MongooseIM cluster
+
+`exml` library used for parsing and emitting of XML messages was upgraded to version 4.0.
+In this new version internal representation of XML elements changed - element attributes
+are stored in a map (previously in a key value list).
+
+This is a disruptive change, and rollback to previous version is not possible.
+
+Parsed XML messages are being sent within MongooseIM cluster between Erlang nodes in internal representation,
+so to understand received messages (Erlang terms), all nodes must have the same code that handle XML elements. This makes a rolling upgrade
+(an upgrade of a cluster node by node) not viable for this release.
+
+One solution is to stop the whole MongooseIM cluster, upgrade and start again.
```

Comment:
Small suggestion: to avoid a "wall of text", we could put the three solutions in a list (numbered or bullet).